opensrp-core/src/main/java/org/smartregister/repository/helper/OpenSRPDatabaseErrorHandler.java

```diff
@@ -3,32 +3,51 @@
 import net.sqlcipher.DatabaseErrorHandler;
 import net.sqlcipher.database.SQLiteDatabase;
 
+import java.io.File;
+
 import timber.log.Timber;
 
 /**
  * Created by ndegwamartin on 18/07/2020.
  */
 public class OpenSRPDatabaseErrorHandler implements DatabaseErrorHandler {
 
-    private final String TAG = getClass().getSimpleName();
-
     /**
      * defines the default method to be invoked when database corruption is detected.
-     *
      * @param dbObj the {@link SQLiteDatabase} object representing the database on which corruption
-     *              is detected.
+     * is detected.
      */
     public void onCorruption(SQLiteDatabase dbObj) {
-        Timber.e("Corruption reported by sqlite on database, db file path: %s", dbObj.getPath());
+        // NOTE: Unlike the AOSP, this version does NOT attempt to delete any attached databases.
+        // TBD: Are we really certain that the attached databases would really be corrupt?
+        Timber.e("Corruption reported by sqlite on database, deleting: %s", dbObj.getPath());
 
         if (dbObj.isOpen()) {
-            Timber.e("Database object for corrupted database is already open, closing");
+            Timber.e( "Database object for corrupted database is already open, closing");
 
             try {
                 dbObj.close();
             } catch (Exception e) {
+                /* ignored */
                 Timber.e(e, "Exception closing Database object for corrupted database, ignored");
             }
         }
+
+        deleteDatabaseFile(dbObj.getPath());
```

Comment:
Is it possible to make the same call the default error handler calls instead of implementing our own?

Reply:
`deleteDatabaseFile` is a **private** method in default class. So even if we override it we won't be able to call it. Also, we will not be able to log any error that occurred while deletion to `Timber`.